### PR TITLE
ci: automatic submodule dependency update

### PR DESCRIPTION
**Pull request by vaind**
See https://github.com/getsentry/sentry-unity/issues/788

sample PR [in the Unity repo](https://github.com/getsentry/sentry-unity/pull/810)

#skip-changelog

### CODE REVIEW

File: .github/workflows/update-deps.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+on:
+  # Run every day.
+  schedule:
+    - cron: '0 3 * * *'
+  # And on on every PR merge so we get the updated dependencies ASAP, and to make sure the changelog doesn't conflict.
+  push:
+    # TODO uncomment
+    # branches:
+    #   - main
```
Comment (marandaneto) [inline .github/workflows/update-deps.yml:11]: Do we need to do something here?

**Comment by marandaneto** [comment]
@vaind is it still a draft or ready to be merged?

**Comment by bruno-garcia** [comment]
@vaind do we get the button to Approve the CI run? If so, that's probably OK.

File: .github/workflows/update-deps.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  native:
+    uses: getsentry/github-workflows/.github/workflows/updater.yml@v1
+    with:
+      path: sentry-android-ndk/sentry-native
+      name: Native SDK
+    secrets:
+      # If a custom token is used instead, a CI would be triggered on a created PR.
+      # api_token: ${{ secrets.CI_DEPLOY_KEY }}
+      api_token: ${{ github.token }}
```
Comment (bruno-garcia) [inline .github/workflows/update-deps.yml:21]:
```suggestion
      api_token: ${{ secrets.CI_DEPLOY_KEY }}
```

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**